cmake/Hunter/config.cmake

```diff
@@ -1,18 +1,33 @@
 
 hunter_config(
     spdlog
-    VERSION 1.8.0-p1
+    VERSION 1.9.2-p0
 )
 
 hunter_config(
     lmdb
     VERSION 0.9.21-p2
 )
 
-hunter_config(
-    OpenSSL
-    VERSION 1.1.1j
-)
+# TODO: Update to 1.1.1q (latest release of OpenSSL) once hunter supports it
+# Then we can remove hacky version for Apple ARM64
+if (APPLE)
```

Comment:
Sorry for getting around to this late... I meant to review it some time ago and then forgot!  I just happened to see it in the backlog...

Anyway, we would need more than just an 'if (APPLE)' check here, since we still need to be able to target x86_64 macOS hardware.  This currently will set the compile architecture to arm64 for both.

Ideally, we could just set `CMAKE_OSX_ARCHITECTURES` to 'arm64;x86_64' and have cmake compile a universal binary for us (assuming we're using the Xcode generator), but I've run into some problems with this involving homebrew that are a nuisance to figure out.  Basically, for any dependencies from homebrew, you need to have an entirely separate installation from the default for another architecture.  There's no way to install libraries for two different architectures from the same installation.  And then this cascades into library path issues (ie, look in one place for x86_64 libraries, and a different one for arm64).  I also ran into issues with Hunter itself when I tried to use two different architectures at the same time.

From what I can tell, it's much easier to build universal binaries from Apple's arm64 arch than it is to build them from x86_64 arch.  I got so frustrated with various compiler and linker errors that I ended up throwing all of it away and figured I'd try again later.

A different option that avoids the whole universal binary issue would be to have cmake ahead of time determine which architecture it is targetting.  This can use the value of CMAKE_OSX_ARCHITECTURES if it is set.  However, if it's not set, some additional check would be needed (maybe using some invocation of `uname -m` in a cmake `execute_process` to set the default arch in some variable, and then that could be checked here in addition to the APPLE variable).

I'm open to other ideas for supporting both arch or resources on building universal binaries from x86_64 (which is our CI arch at the moment).